src/function/filter.ts

```diff
@@ -0,0 +1,44 @@
+import { Person } from "../types";
+import { FILTER } from "../types/filters";
+
+export function filter(
+    people: Person[],
+    filterSex: FILTER,
+    query: string,
+    centuries: string[]
+  ): Person[] {
+    let filteredPeople = [...people];
+  
+    switch (filterSex) {
+      case FILTER.MALE:
+        filteredPeople = filteredPeople.filter(person => person.sex === 'm');
+        break;
+  
+      case FILTER.FEMALE:
+        filteredPeople = filteredPeople.filter(person => person.sex === 'f');
+        break;
+  
+      case FILTER.ALL:
+      default:
+        break;
+    }
+  
+    if (query) {
+      filteredPeople = filteredPeople.filter(person => {
+        return (
+          person.name.toLowerCase().includes(query.toLowerCase()) ||
+          person.motherName?.toLowerCase().includes(query.toLowerCase()) ||
+          person.fatherName?.toLowerCase().includes(query.toLowerCase())
```

Comment:
Ensure that the `toLowerCase()` method is used consistently to handle case-insensitive searches. This is correctly implemented here, but remember to handle potential null values for `motherName` and `fatherName` as you have done.